README-en.md

```diff
@@ -14,3 +14,45 @@ A lexer is a tool that allows us to analyze a sequence of characters and extract
 
 It was used as a programming language *__Haskell__* and as a generator tool for lexicographical analyzers *__Alex__*.
 The use of Alex was fundamental, as we focused on only determining regular expressions to extract the tokens from the file.
+
+#### Brief explanation
+
+Alex provides several wrappers, but the one of our interest was `posn` because it provides the same functionality as the` basic` but adding the column and row of each detected token.
+
+To implement the `show` of each token, a type of data` TkObject` was created that can be seen as an ordered pair whose first coordinate is a token and the second is its `AlexPosn`. Thus, it was avoided having to write a `show` for each token.
+
+To filter errors, a token called `TkErr` was created. When we invoke `alexScanTokens` we have an arrangement of `TkObject`'s. To see if there is an error, we only filter the TkObjects and if the resulting fix is empty, there are no errors and each token is printed in the described format. Otherwise, only errors are printed.
```

Comment:
```suggestion
To filter errors, a token called `TkErr` was created. When we invoke `alexScanTokens` we have an array of `TkObject`'s. To see if there is an error, we only filter the TkObjects and if the resulting array is empty, there are no errors and each token is printed in the described format. Otherwise, only errors are printed.
```